src/test/java/com/rybka/service/export/CsvExportServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package com.rybka.service.export;
+
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import com.rybka.configuration.ExportProperty;
+import com.rybka.exception.ExportFailureException;
+import com.rybka.model.CurrencyHistory;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+public class CsvExportServiceTest {
+
+    private final CsvMapper csvMapper = new CsvMapper();
+    @Mock
+    private ExportProperty exportProperty;
+
+    @Test
+    public void testOnProperExport() {
+
+        // given
+        var testFolder = "src/test/resources/";
+        var testType = "csv";
+        var currencyBase = "USD";
+        var currencyTarget = "EUR";
+        var currencyRate = 0.88;
+        var currencyAmount = 4d;
+        var currencyTotal = currencyRate * currencyAmount;
+        var currencyHistory = new CurrencyHistory(currencyBase, currencyTarget, currencyAmount, currencyRate, currencyTotal);
+        currencyHistory.setDate(Calendar.getInstance().getTime());
+        var testList = List.of(currencyHistory);
+        var csvExportService = new CsvExportService(csvMapper, exportProperty);
+
+        when(exportProperty.getFolder()).thenReturn(testFolder);
+        when(exportProperty.getType()).thenReturn(testType);
+
+        // when
+        csvExportService.export(testList);
+        Path testFilePath = findSavedFile(testFolder, testType);
+        var actualResult = readSavedFile(testFilePath);
+
+        // then
+        verify(exportProperty).getType();
+        verify(exportProperty).getFolder();
+
+        Assertions.assertEquals(currencyRate, actualResult.getRate());
+
+        removeSavedFile(testFilePath);
+    }
+
+    @Test
+    public void testOnExportException() {
+
+        // given
+        var testType = "csv";
+        var currencyBase = "USD";
+        var currencyTarget = "EUR";
+        var currencyRate = 0.88;
+        var currencyAmount = 4d;
+        var currencyTotal = currencyRate * currencyAmount;
+        var currencyHistory = new CurrencyHistory(currencyBase, currencyTarget, currencyAmount, currencyRate, currencyTotal);
+        currencyHistory.setDate(Calendar.getInstance().getTime());
+        var testList = List.of(currencyHistory);
+        var csvExportService = new CsvExportService(csvMapper, exportProperty);
+
+        when(exportProperty.getFolder()).thenThrow(ExportFailureException.class);
```

Comment:
Are you sure getFolder() can throw `ExportFailureException` ?

Reply:
fixed